cpp/lib/src/Subfields.cc

```diff
@@ -71,76 +72,88 @@ std::string Subfields::getFirstSubfieldValue(const char subfield_code) const {
     const auto begin_end(getIterators(subfield_code));
     if (begin_end.first == begin_end.second)
         return "";
-    return begin_end.first->second;
+    return begin_end.first->value_;
+}
+
+
+void Subfields::addSubfield(const char subfield_code, const std::string &subfield_data) {
+    auto insert(begin());
+    for (/* empty */; insert != end() && insert->code_ <= subfield_code; ++insert);
+    subfields_.emplace(insert, subfield_code, subfield_data);
 }
 
 
 void Subfields::replace(const char subfield_code, const std::string &old_value, const std::string &new_value) {
     bool found(false);
 
-    const std::pair<Iterator, Iterator> begin_end(getIterators(subfield_code));
+    const auto begin_end(getIterators(subfield_code));
     for (Iterator code_and_value(begin_end.first); code_and_value != begin_end.second; ++code_and_value) {
-        if (code_and_value->second == old_value) {
+        if (code_and_value->value_ == old_value) {
             found = true;
-            code_and_value->second = new_value;
+            code_and_value->value_ = new_value;
         }
     }
 
     if (not found)
-        throw std::runtime_error("Unexpected: tried to replace \"" + old_value + "\" with \"" + new_value + "\" in subfield '"
-                                 + subfield_code + "' but did not find the original value!");
+        throw std::runtime_error(
```

Comment:
Maybe add in the function name "Subfields::replace" as part of the error message.